### ISSUE TITLE
helpful error message when the debuggee is incompatible with the host OS/arch

### ISSUE THREAD

**Issue by hyangah** [go-delve/delve#2058]
This is a feature request (based on https://github.com/microsoft/vscode-go/issues/2696.

If the program to debug is not executable on the current `GOHOSTOS/GOHOSTARCH`
so `dlv` cannot launch, I'd like to see more helpful message than the current message.

1. What version of Delve are you using (`dlv version`)? 1.4.0
2. What version of Go are you using? (`go version`)? go1.14.x
3. What operating system and processor architecture are you using? darwin/amd64
4. What did you do?
Basically, `GOOS=linux dlv debug` on mac OS. Accidentally cross-compiled. :-)
Similar issue when `dlv exec` when the target binary isn't executable.
 
5. What did you expect to see?
An error message indicating the target binary is not executable.
For example, `go run` command outputs `exec format error`.
Attempt to run such incompatible binaries directly outputs  `cannot execute binary file`.

6. What did you see instead?
`could not launch process: stub exited while waiting for connection: exit status 0`
which is vague.
Or, is this (incompatible or corrupted executable) the only case dlv would fail to launch the process with the error message? (If so, maybe rephrase the error message?)

**Comment by derekparker** [comment]
Have you tried with Delve built from master? I sent in a patch to fix this issue a month ago: https://github.com/go-delve/delve/pull/1979.

Closing for now, we can reopen if this is still an issue.

**Comment by hyangah** [comment]
Yes, it works! Thanks!
When will be the new version release? We were with v1.4.0 quite some time.
